### PR TITLE
Deactivate webhook before deletion

### PR DESCRIPTION
**Pull request by jakubkuc**
Discussion and reviews for this cherry-picks are in #11239 

I want to merge this change because it prevents issue where using WebhookDelete mutation while tasks that create EventDeliveryAttempt are queued resulted in IntegrityError.

The most simple resolution was to deactivate Webhook before deleting it in mutation. send_webhook_request_async task is checking if webhook is_active so this should prevent this situation.

Tasks:
I want to add tests after a discussion regarding edge-case.

Things to discuss:
⚠️ There is a small window for this solution to fail - when Webhook deactivation will happen between the is_active check and attempt creating/updating in send_webhook_request_async task. I'm not quite sure how to handle this case. Should I add more webhook activity checks inside send_webhook_request_async task or maybe catch this edge-case exception to fail gracefully? Any ideas?

Second thing: Do we want to do the sam analogically for deleting App? (deactivate webhook in app deletion)

https://github.com/saleor/saleor/issues/11227

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

File: saleor/graphql/webhook/mutations.py [hunk 1]
```diff
@@ -232,7 +233,10 @@ class Arguments:
         id = graphene.ID(required=True, description="ID of a webhook to delete.")
 
     class Meta:
-        description = "Deletes a webhook subscription."
+        description = (
+            "Deletes a webhook subscription. Deletion might fail due to "
+            "webhook delivery in progress. In this case deactivates webhook"
```
Comment (maarcingebala) [inline saleor/graphql/webhook/mutations.py:238]: Redacted it to be more descriptive. This is our docs so needs to be as clear as possible to the users. 

```suggestion
            "Delete a webhook. Before the deletion, the webhook is deactivated to "
            "pause any deliveries that are already scheduled. The deletion might fail "
            "if delivery is in progress. In such a case, the webhook is not deleted "
            "but remains deactivated."
```

Comment (maarcingebala) [reply saleor/graphql/webhook/mutations.py:238]: Will need schema regeneration after it's updated.

**Review by korycins — APPROVED** [review]
